### PR TITLE
Allow RestHandler to pause execution

### PR DESCRIPTION
**Pull request by graetzer**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/564/

File: arangod/GeneralServer/RestHandler.h [hunk 1]
```diff
@@ -64,25 +63,31 @@ class RestHandler : public std::enable_shared_from_this<RestHandler> {
...
+    TRI_ASSERT(_state == HandlerState::PAUSED);
+    runHandlerStateMachine();
   }
-
+  
   /// Execute the rest handler state machine
-  int runHandler(std::function<void(rest::RestHandler*)> cb);
+  void runHandler(std::function<void(rest::RestHandler*)> const& cb) {
+    TRI_ASSERT(_state == HandlerState::PREPARE);
+    _callback = cb;
```
Comment (mpoeter) [inline arangod/GeneralServer/RestHandler.h]: You could take `cb` per value and move into `_callback` to avoid copies.

**Comment by jsteemann** [comment]
Does not compile on some platforms:
```
/work/ArangoDB/arangod/Scheduler/Scheduler.cpp: In member function 'bool arangodb::rest::Scheduler::queue(std::unique_ptr<arangodb::Job>)':
/work/ArangoDB/arangod/Scheduler/Scheduler.cpp:366:3: error: incomplete type 'arangodb::RequestStatistics' used in nested name specifier
   RequestStatistics::SET_QUEUE_START(job->_handler->statistics(), queueSize);
   ^~~~~~~~~~~~~~~~~
```

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/566/

**Pull request merged**